src/nvim/ex_cmds.c

```diff
@@ -3999,22 +4050,30 @@ static buf_T *do_sub(exarg_T *eap, proftime_T timeout)
 
   // Show 'inccommand' preview if there are matched lines.
   buf_T *preview_buf = NULL;
+  size_t subsize = preview_lines.subresults.size;
   if (preview && !aborting()) {
     if (got_quit) {  // Substitution is too slow, disable 'inccommand'.
       set_string_option_direct((char_u *)"icm", -1, (char_u *)"", OPT_FREE,
                                SID_NONE);
-    } else if (*p_icm != NUL && matched_lines.size != 0 && pat != NULL) {
+    } else if (*p_icm != NUL &&  pat != NULL) {
+      if (pre_src_id == 0) {
+        // Get a unique new src_id, saved in a static
+        pre_src_id = bufhl_add_hl(NULL, 0, -1, 0, 0, 0);
+      }
+      if (pre_hl_id == 0) {
+        pre_hl_id = syn_check_group((char_u *)"Substitute", 13);
```

Comment:
This should be `10`, right? I'll change it to `sizeof("Substitute") - 1`.

Reply:
Ah yeah, thats a leftover from "IncSubstitute" I guess :)